docs/tutorials/intro.md

```diff
@@ -209,8 +238,41 @@ fn main() {
 }
 ```
 
+You will also need to use the `with_buffer_provider` constructors, e.g. `DateTimeFormatter::try_new_with_buffer_provider()`.
+
 The ICU4X repository has test data checked in tree in `provider/testdata/data`, however it is recommended one generate data on their own as described in the [next section](#generating-data). Under the hood, `icu_testdata` is simply loading this data.
 
+Production users are recommended to use `BlobDataProvider` from `icu_provider_blob`, which allows a binary blob of (usually `postcard` format) data to be loaded from memory. This data provider provides the flexibility of controlling where the data is stored; allowing for data to even be loaded lazily over the network.
+
+Here is a full example:
+
+```rust
+use icu::locid::locale;
+use icu::calendar::DateTime;
+use icu::datetime::{DateTimeFormatter, options::length};
+use icu_provider_fs::FsDataProvider;
+
+fn main() {
+    let date = DateTime::try_new_gregorian_datetime(2020, 10, 14, 13, 21, 28)
+        .expect("Failed to create a datetime.");
+
+    let options = length::Bag::from_date_time_style(length::Date::Long, length::Time::Medium);
+
+    let provider = FsDataProvider::try_new("/path/to/data")
+       .expect("Failed to initialize Data Provider.");
+
+    let date = date.to_any();
+
+    let dtf = DateTimeFormatter::try_new_with_buffer_provider(&provider, &locale!("ja").into(), options.into())
+        .expect("Failed to initialize DateTimeFormatter");
+
+    let formatted_date = dtf.format(&date).expect("Formatting should succeed");
```

Comment:
`optional, n-b`: first initialize `dtf`, then `date` - my brain was truing to understand how the constructor uses `date`.